### PR TITLE
Add patch for Rails 5.2.x

### PR DESCRIPTION
**Pull request by n-rodriguez**

### CODE REVIEW

**Comment by webm0nk3y** [comment]
It would be wonderful if this could be accepted.

**Comment by n-rodriguez** [comment]
@jonkessler 
Hi! Rails 5.2 is out :) http://weblog.rubyonrails.org/
I think we could drop support of Rails 4.0.x and 4.1.x in this new release. What do you think?

File: .travis.yml [hunk 1]
```diff
@@ -53,66 +53,96 @@ gemfile:
...
+      gemfile: gemfiles/rails_5.1.5.gemfile
+    - rvm: 2.4.3
+      gemfile: gemfiles/rails_5.2.0.gemfile
   exclude:
     # These Rails versions are broken with the latest Ruby versions.
     # They are fixed in the next version:
     # https://github.com/rails/rails/blob/v4.0.6/activerecord/CHANGELOG.md
     # https://github.com/rails/rails/blob/v4.1.2/activerecord/CHANGELOG.md
-    - rvm: 2.2.7
+    - rvm: 2.2.9
```
Comment (bboe) [inline .travis.yml]: Are these exclusions still necessary?

**Comment by bboe** [comment]
Given the addition of the `polymorphic_name` method, I'm wondering if this package is even necessary in rails5. Or to ease transition, perhaps we need only overwrite that method going forward.

https://github.com/rails/rails/blob/d35875b7a3f559155a9378cbe9203b0b8ea580f9/activerecord/lib/active_record/inheritance.rb#L159

**Comment by n-rodriguez** [comment]
Actually this gem is useless for Rails 5.2 but not for Rails 5.1 and before.
Rails 5.1 : https://github.com/rails/rails/blob/5-1-stable/activerecord/lib/active_record/inheritance.rb
Rails 5.2 : https://github.com/rails/rails/blob/5-2-stable/activerecord/lib/active_record/inheritance.rb

**Comment by bboe** [comment]
@n-rodriguez would you consider updating such that for rails 5.2 only the `polymorphic_name` method is updated? It'd be nice to know that all the tests pass with such a simple change. In a subsequent pull request we could even go so far as to output a deprecation warning suggesting how to make it so this gem isn't necessary.

**Comment by n-rodriguez** [comment]
I'm on it ;)

**Comment by n-rodriguez** [comment]
2 tests are broken :

```ruby
  1) Failure:
TestStoreBaseStiClass#test_polymorphic_has_many_through_with_double_sti_on_join_model [/home/nicolas/PROJECTS/GITHUB/store_base_sti_class/test/test_store_base_sti_class.rb:145]:
Expected: "SpecialPost"
  Actual: "Post"

  2) Failure:
TestStoreBaseStiClass#test_has_many_through_polymorphic_has_one [/home/nicolas/PROJECTS/GITHUB/store_base_sti_class/test/test_store_base_sti_class.rb:116]:
--- expected
+++ actual
@@ -1 +1 @@
-[#<Tagging id: 14, tag_id: 20, polytag_id: nil, polytag_type: nil, taggable_type: "Post", taggable_id: 21>, #<Tagging id: 15, tag_id: 20, polytag_id: nil, polytag_type: nil, taggable_type: "SpecialPost", taggable_id: 22>]
+[#<Tagging id: 14, tag_id: 20, polytag_id: nil, polytag_type: nil, taggable_type: "Post", taggable_id: 21>]
```

**Comment by bboe** [comment]
> 2 tests are broken

Can we isolate the changes that are needed to get those to work?

**Comment by n-rodriguez** [comment]
> Thanks for looking into that. Can we isolate the changes that are needed to get those to work?

Done!

**Comment by bboe** [comment]
Awesome. Taking a look.

**Comment by bboe** [comment]
Thanks for submitting! You're awesome. 🎆

**Comment by n-rodriguez** [comment]
You're welcome :)

**Comment by bboe** [comment]
Released 1.2.0.

**Pull request merged**